### PR TITLE
changing logrotate config to keep more rails logs and less nginx logs

### PR DESCRIPTION
**Pull request by gagou9**
voilà

### CODE REVIEW

**Comment by jfly** [comment]
Thanks for doing this @gagou9! Before we merge this up, could you provide an estimate on how much space you expect the logs to take up now?

File: chef/site-cookbooks/wca/recipes/default.rb [hunk 1]
```diff
@@ -132,11 +124,11 @@
...
-  frequency WCA_LOGROTATE_FREQUENCY
-  maxsize WCA_LOGROTATE_MAXSIZE
-  size WCA_LOGROTATE_SIZE
-  rotate WCA_LOGROTATE_ROTATE
-  path "#{repo_root}/WcaOnRails/log/production.log"
+  path		"#{repo_root}/WcaOnRails/log/production.log"
+  size		"512M"
+  maxage	90
+  postrotate	"[ ! -f /home/cubing/worldcubeassociation.org/WcaOnRails/pids/unicorn.pid ] || kill -USR1 `cat /home/cubing/worldcubeassociation.org/WcaOnRails/pids/unicorn.pid`"
+  options	['nodelaycompress']
```
Comment (jfly) [inline chef/site-cookbooks/wca/recipes/default.rb]: Nit: For consistency with the rest of this file (and our codebase), let's not align things like this. One space should be consistent with what we do elsewhere.

Comment (gagou9) [inline chef/site-cookbooks/wca/recipes/default.rb]: Ho, I find it much easier to read this way, but ok!

**Comment by gagou9** [comment]
About the size of the logs, the nginx logs will be rotated when they reach 512M, and the rotated logs will be kept until they are 30 days old. Currently, we have access.log.1 that is 512M, and has logs from the 7th to the 18th of september, so we can guess we will have up to 3 rotated files, which are 45M each after compression, so that brings us to up to 700M for access.log (the non rotated one that will reach 512M at some point plus the rotated ones, 45M each). The error.log log will probably last much longer and might even not been rotated, because I'm not sure whether the maxage option will say "hey, error.log has to be rotated and is older than 30 days, let's delete it" or "hey, error.log has to be rotated, let's do that and check its age at next rotation.". If it's the second option, then error.log* will reach 600M total (~512M plus 45M)
So that's 1.3G for nginx.

For production.log, it reaches 512M every day, so it will be rotated every day, so we will have 90 rotated and compressed files being about 80M each (some are less, some are more), which makes a total of 7.2G plus the uncompressed one that might grow up to about 3G.

So we have a total of about 13G for nginx and production.log, if the amount of logged things doesn't grow.

That's all if I didn't make a mistake ;)

**Comment by jfly** [comment]
LGTM! Can you merge this up and run chef on the production server?

**Comment by gagou9** [comment]
Hoy,

So, I merged that pull request, then it got merged to production I don't know how.
Then I didn't know how to run chef on the production server, so I only ran "pull_latest" and didn't try that `chef-solo -o 'role[wca]'` from  https://github.com/cubing/worldcubeassociation.org/blob/master/scripts/wca-bootstrap.sh#L110
I connected to staging, ran `pull_latest rebuild_regs rebuild_rails`
Staging seems to work, but when I ran `sudo chef-solo -o 'role[wca]'` chef didn't change the logrotate config.
I ran it a second time and it worked. I then manualy changed "maxsize" in both rails-wca and nginx-wca logrotate config files, and ran `logrotate /etc/logrotate.conf` and it successfully rotated the logs.
I will now run `sudo chef-solo -o 'role[wca]'` on production.

**Comment by gagou9** [comment]
Okay, it worked ;) (apparently)

**Comment by jfly** [comment]
Sounds good. Thanks for the detailed description of what you did!

**Comment by gagou9** [comment]
For your info, I also just ran `gzip /var/log/nginx/access.log.1 ./worldcubeassociation.org/WcaOnRails/log/production.log.1`. These files were not compressed and I wonder how logrotate would have handled this.

**Pull request merged**